pandas_gbq/gbq.py

```diff
@@ -872,82 +851,58 @@ def _parse_entry(field_value, field_type):
     elif field_type == 'BOOLEAN':
         return field_value == 'true'
     return field_value
+    
 
-
-def read_gbq(query, project_id=None, index_col=None, col_order=None,
-             reauth=False, verbose=True, private_key=None,
-             auth_local_webserver=False, dialect='legacy', **kwargs):
-    r"""Load data from Google BigQuery.
+def read_gbq(query, project_id=None, index_col=None, col_order=None, verbose=True,
+             private_key=None, dialect='legacy', configuration=None, **kwargs):
+    r"""Load data from Google BigQuery using google-cloud-python
 
     The main method a user calls to execute a Query in Google BigQuery
     and read results into a pandas DataFrame.
 
-    Google BigQuery API Client Library v2 for Python is used.
+    The Google Cloud library is used.
     Documentation is available `here
-    <https://developers.google.com/api-client-library/python/apis/bigquery/v2>`__
-
-    Authentication to the Google BigQuery service is via OAuth 2.0.
+    <https://googlecloudplatform.github.io/google-cloud-python/stable/>`__
 
-    - If "private_key" is not provided:
-
-      By default "application default credentials" are used.
-
-      If default application credentials are not found or are restrictive,
-      user account credentials are used. In this case, you will be asked to
-      grant permissions for product name 'pandas GBQ'.
-
-    - If "private_key" is provided:
-
-      Service account credentials will be used to authenticate.
+    Authentication via Google Cloud can be performed a number of ways, see:
+    <https://googlecloudplatform.github.io/google-cloud-python/stable/google-cloud-auth.html>
+    The easiest is to download a service account JSON keyfile and point to it using 
+    an environment variable:
+    `$ export GOOGLE_APPLICATION_CREDENTIALS="/path/to/keyfile.json"`
 
     Parameters
     ----------
     query : str
         SQL-Like Query to return data values
-    project_id : str
+    project_id : str (optional)
         Google BigQuery Account project ID.
     index_col : str (optional)
         Name of result column to use for index in results DataFrame
     col_order : list(str) (optional)
         List of BigQuery column names in the desired order for results
         DataFrame
-    reauth : boolean (default False)
-        Force Google BigQuery to reauthenticate the user. This is useful
-        if multiple accounts are used.
     verbose : boolean (default True)
         Verbose output
     private_key : str (optional)
-        Service account private key in JSON format. Can be file path
-        or string contents. This is useful for remote server
-        authentication (eg. jupyter iPython notebook on remote host)
-    auth_local_webserver : boolean, default False
```

Comment:
You should be able to keep this. The existing auth code on master should work seamlessly with both the Google API client libs and the Google Cloud client libs.

There's an example of using the local web server for auth here: https://cloud.google.com/bigquery/docs/authentication/end-user-installed